src/BCEngineMCMC.cxx

```diff
@@ -1005,32 +1008,91 @@ bool BCEngineMCMC::LoadParametersFromTree(TTree* partree, bool loadObservables)
         Par.FillHistograms(p_fill_1d, p_fill_2d);
         Par.SetNbins(p_nbins);
         AddParameter(Par);
-        if (p_scale) {
-            if (p_nchains != scales.size())
-                throw std::runtime_error("BCEngineMCMC::LoadParametersFromTree: nchain mismatch");
-            for (unsigned c = 0; c < p_nchains; ++c)
-                scales[c].push_back(*p_scale[c]);
+    }
+
+    // load user-defined observables
+    if (loadObservables) {
+        fObservables = BCObservableSet();
+        // loop over entries with "parameter" branch = false
+        for (unsigned i = 0; partree->GetEntryNumberWithIndex(0, i) >= 0; ++i) {
+            partree->GetEntryWithIndex(0, i);
+            BCObservable Obs(p_name, p_lowerlimit, p_upperlimit, p_latexname);
+            Obs.SetPrecision(p_precision);
+            Obs.FillHistograms(p_fill_1d, p_fill_2d);
+            Obs.SetNbins(p_nbins);
+            AddObservable(Obs);
+        }
+    }
+
+    partree->ResetBranchAddresses();
+
+    return true;
+}
+
+// --------------------------------------------------------
+void BCEngineMCMC::LoadMCMCParameters(TTree& partree)
+{
+    if (partree.GetEntries() < 1)
```

Comment:
The two functions that do this each check for different essential branches---only those needed for the loading of what that function loads. Since the functions aren't always called together (`LoadMCMCParameters` is not called when simply loading in data for remarginalization) we can't have one consolidated function---it would overcheck.

Reply:
ok